### DISCUSSION TITLE
Plugin to overwrite "nginx-app-template-source" trigger, sigil templating fails (Using Go)

### DISCUSSION THREAD

**Discussion by dewey** [dokku/dokku#6991] [Q&A] [answered]
Hey, I'm working on a small personal plugin to overwrite the "nginx-app-template-source" trigger as suggested [in another post](https://github.com/dokku/dokku/discussions/6688#discussioncomment-8743129) here. The code of [my plugin is on GitHub](https://github.com/dewey/dokku-nginx-override-by-app) and it's very simple.

It has two features: A command to move a config to a specific place and implementing the trigger to return that path based on the app name.

It seems to work except that the templating step is failing as the variables are not set. That makes the `split` function (provided by sigil) used in the [default nginx.conf.sigil](https://github.com/dokku/dokku/blob/master/plugins/nginx-vhosts/templates/nginx.conf.sigil) fail if an empty string is passed into it. It feels like somehow there's a step missing that populates all the variables but I'm not sure why (and how?) that would need to be done in the plugin as that one is just supplying one string (the path to the custom template location).

```
-----> Creating https nginx.conf
       Enabling HSTS
template: nginx.conf.sigil:1:76: executing "nginx.conf.sigil" at <split " ">: error calling split: split must be given a string
 !      !     No healthchecks found in app.json for web process type
template: nginx.conf.sigil:1:76: executing "nginx.conf.sigil" at <split " ">: error calling split: split must be given a string
```

Here's a full `ps:rebuild` [with trace](https://gist.github.com/dewey/6ccd5ec9b406c68fa74d17ca294ea43e), and [one without](https://gist.github.com/dewey/bd4e405393102798179f64e2d66a6498) for easier readability.

Happy to hear other feedback too, as this is my first plugin.

Thanks!

**Comment by josegonzalez** [comment, marked as answer]
Your bug is that you are ignoring the second argument. That argument controls which template to render. In this case, you are telling Dokku to use the `nginx.conf.sigil` to render `hsts-config`. You should only respond with a template when the argument is `app-config`.

**Reply by dewey** [reply]
That was it, thank you so much! Now everything is working very nicely! 🙌
